symbolic-debuginfo/src/plist.rs

```diff
@@ -0,0 +1,155 @@
+//! Apple PropertyList support.
+//!
+//! For MachO objects
+
+use std::collections::HashMap;
+use std::error::Error as StdError;
+use std::fmt;
+use std::io::Cursor;
+
+use elementtree::Element;
+use thiserror::Error;
+
+use symbolic_common::{DebugId, ParseDebugIdError};
+
+/// The error type for handling a [`PList`].
+#[derive(Debug, Error)]
+#[error("{kind}")]
+pub struct PListError {
```

Comment:
it's kind of because plist is more general i put it here.  it doesn't actually try to reject plists not related to dsyms, only has an `.is_bcsymbol_mapping()` to determine if it can be used.

I imagine when we bump into plist support issues we'd move to an external crate for plist parsing.  Which means maybe this should not end up in the public API like this.  I think moving it to the new bcsymbolmap module might make more sense.  Maybe a more simple ‘OriginalUUIDMapping` with `::parse()`, `::test()` and `::original_uuid()` methods as that hides more of the implementation?